### PR TITLE
Merge SilkX into develop/3.0

### PR DESCRIPTION
**Pull request by Perksey**
As discussed in the Working Group meeting.

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                dotnet-version: 6.0.300
+            - name: Generate Report
+              uses: danielpalme/ReportGenerator-GitHub-Action@5.1.9
+              with:
+                reports: './coverage/**/**/coverage.cobertura.xml'
+                targetdir: './coveragereport'
+                reporttypes: 'Cobertura'
+                verbosity: 'Info'
+                tag: '${{ github.run_number }}_${{ github.run_id }}'
+            - uses: 5monkeys/cobertura-action@master
```
Comment (Perksey) [inline .github/workflows/dotnet.yml:73]: Allow fail, sometimes we can't post the results as a check run.

File: Directory.Build.targets [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  -->
+
+  <!-- Settings that append the existing setting value -->
+  <PropertyGroup>
+    <DefineConstants>$(DefineConstants);$(OS)</DefineConstants>
+  </PropertyGroup>
+
+  <!-- Package versions for package references across all projects -->
+  <ItemGroup>
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.1.1" />
```
Comment (Perksey) [inline Directory.Build.targets:20]: CPVM?

File: documentation/deprecation-notices/SilkManager.md [hunk 3]
```diff
@@ -1,23 +0,0 @@
-# SilkManager
```
Comment (Perksey) [inline documentation/deprecation-notices/SilkManager.md:1]: Probably worth bringing the documentation folder back...

**Comment by Perksey** [comment]
@dotnet/silk-dotnet for your perusal

**Comment by Perksey** [comment]
NOTE: DO NOT SQUASH THIS ONE! Merge only please.

File: Directory.Build.targets (hunk 2 above)
Comment (Perksey) [reply Directory.Build.targets:20]: Will do this at another time.

**Review by ThomasMiz — APPROVED** [review]

**Pull request merged**